_dev/apps/ui/src/store/modules/product-feed/state.ts

```diff
@@ -34,15 +34,25 @@ export type ProductFeedReport = {
 }
 export interface ProductFeedSettings {
   shippingSetup: ShippingSetupOption|null;
-  estimateCarriers: CustomCarrier[];
-  shippingSettings: ShopShippingInterface[];
-  rate: RateType|null;
-  deliveryDetails: DeliveryDetail[];
+  estimateCarriers?: CustomCarrier[];
+  shippingSettings?: ShopShippingInterface[];
```

Comment:
These variable have been made mandatory in the vue store to prevent issues with the reactiveness of the UI (going from undefined to a value prevents us to see the form being updated).

If you made this change for the API payload, don't you think it should go to its own type?

Reply:
ok. so you're answering the question I asked in another comment.
As these fields must always be present, that means I need to add a default value.